### PR TITLE
snappy: implement first pass at config snippets

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: snappy/overlord.go [hunk 1]
```diff
@@ -342,6 +344,49 @@ func (o *Overlord) SetActive(s *SnapPart, active bool, meter progress.Meter) err
...
+
+	var m map[string]map[string]interface{}
+	if err := yaml.Unmarshal([]byte(cfg), &m); err != nil {
+		return "", fmt.Errorf("configuration succeeded but can't unmarshal result: %v", err)
+	}
+
+	v, ok := m["config"][name]
+	if !ok {
+		// is this an error?
+		// config succeeded but returned the wrong lind of thing...
```
Comment (zyga) [inline snappy/overlord.go]: s/lind/kind/

File: snappy/overlord.go [hunk 2]
```diff
@@ -342,6 +344,49 @@ func (o *Overlord) SetActive(s *SnapPart, active bool, meter progress.Meter) err
 	return s.deactivate(false, meter)
 }
 
+// ConfigureFromSnippet turns snippet into a snap configuration
+// and calls Configure on the result.
```
Comment (zyga) [inline snappy/overlord.go]: Please explain what the snippet is. I had to run back and forth in this diff to catch that.

**Comment by zyga** [comment]
+1 with the two trivials mentioned

**Comment by chipaca** [comment]
retest this please

**Pull request closed without merging**